qt/widgets/instrumentview/src/InstrumentActor.cpp

```diff
@@ -1206,4 +1206,13 @@ const Mantid::Geometry::DetectorInfo &InstrumentActor::detectorInfo() const {
   else
     return getWorkspace()->detectorInfo();
 }
+
+bool InstrumentActor::canPlotBlocksize() const {
+  Mantid::API::MatrixWorkspace_const_sptr inputWs = getWorkspace();
+  if (inputWs->blocksize() > 1) {
```

Comment:
Does there need to be a try/catch here then for ragged workspaces for now until handled properly?